lib/features/weather/viewmodel/weather_viewmodel.dart

```diff
@@ -14,37 +13,28 @@ YumemiWeather yumemiWeather(YumemiWeatherRef ref) => YumemiWeather();
 @riverpod
 class WeatherViewModel extends _$WeatherViewModel {
   @override
-  WeatherViewModelState build() => const WeatherViewModelState();
+  Result<WeatherViewModelState, Exception> build() => const Result.success(
+        WeatherViewModelState(),
+      );
 
   void fetchWeather({
     required String area,
     required DateTime date,
   }) {
-    final yumemiWeather = ref.read(yumemiWeatherProvider);
-    try {
-      final req = FetchWeatherRequest(
-        area: area,
-        date: date,
-      );
-      final jsonResponse = yumemiWeather.fetchWeather(jsonEncode(req));
-      final weatherResponse = FetchWeatherResponse.fromJson(
-        jsonDecode(jsonResponse) as Map<String, dynamic>,
-      );
-      state = state.copyWith(
-        weatherCondition: weatherResponse.weatherCondition,
-        maxTemperature: weatherResponse.maxTemperature,
-        minTemperature: weatherResponse.minTemperature,
-      );
-    } on YumemiWeatherError catch (e) {
-      switch (e) {
-        case YumemiWeatherError.unknown:
-          throw Exception('不明なエラーが発生しました');
-        case YumemiWeatherError.invalidParameter:
-          throw Exception('パラメータが不正です');
-      }
-    } on FormatException {
-      // jsonDecode() でエラーが発生した場合
-      throw Exception('レスポンスのフォーマットが不正です');
-    }
+    final req = FetchWeatherRequest(
+      area: area,
+      date: date,
+    );
+    final useCase = ref.read(fetchWeatherUseCaseProvider);
+    useCase(req).when(
+      success: (data) => state = Result.success(
+        WeatherViewModelState(
+          weatherCondition: data.weatherCondition,
+          minTemperature: data.minTemperature,
+          maxTemperature: data.maxTemperature,
+        ),
+      ),
+      failure: (e) => state = Result.failure(e),
+    );
```

Comment:
[nits]
若干ですが、このようにしたほうが可読性高まりそうです。

```suggestion
    state = useCase(req).when(
      success: (data) => Result.success(
        WeatherViewModelState(
          weatherCondition: data.weatherCondition,
          minTemperature: data.minTemperature,
          maxTemperature: data.maxTemperature,
        ),
      ),
      failure: Result.failure,
    );
```

Reply:
ありがとうございます
確かに、こちらのほうがきれいな書き方だと感じました。
修正しました 13d691c8a80a40515fca5fa592a41fe783e3b06a